plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/engine/HybridMobileEngineManager.java

```diff
@@ -99,14 +116,105 @@ public HybridMobileEngineManager(HybridProject project){
 	private boolean engineMatches(Engine configEngine, HybridMobileEngine engine){
 		//null checks needed: sometimes we encounter engines without a name or version attribute.
 		if(engine.isManaged()){
-			return configEngine.getName() != null && configEngine.getName().equals(engine.getId()) &&
-				configEngine.getSpec() != null && configEngine.getSpec().equals(engine.getVersion());
+
```

Comment:
We are using a SemVer library which was supposed to handle the version expressions on its next version. I will update the library and that we can move to use it. This below section is not %100 correct.